File: YangDoPlatform/app/src/main/java/se/jbnu/yangdoplatform/SigninActivity.java
package se.jbnu.yangdoplatform;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;


public class SigninActivity extends AppCompatActivity {
    private Button signup;
    private Button login;
    private EditText email_login;
    private EditText pwd_login;
    FirebaseAuth firebaseAuth;
    private FirebaseAuth.AuthStateListener firebaseAuthListener;
    private int RC_SIGN_IN = 123;
    private String TAG = "mainTag";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        firebase_login_check();
        setContentView(R.layout.activity_signin);

        signup = (Button) findViewById(R.id.btn_signup);
        login = (Button) findViewById(R.id.btn_login);

        email_login = (EditText) findViewById(R.id.user_id_input);
        pwd_login = (EditText) findViewById(R.id.user_password_input);
        firebaseAuth = FirebaseAuth.getInstance();



        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = email_login.getText().toString().trim();
                String pwd = pwd_login.getText().toString().trim();

                if (email.length() > 0 && pwd.length() > 0) {
                    firebaseAuth.signInWithEmailAndPassword(email, pwd)
                            .addOnCompleteListener(SigninActivity.this, new OnCompleteListener<AuthResult>() {
                                @Override
                                public void onComplete(@NonNull Task<AuthResult> task) {
                                    if (task.isSuccessful()) {//성공했을때
                                        Toast.makeText(SigninActivity.this, "로그인 성공", Toast.LENGTH_SHORT).show();
                                        Intent intent = new Intent(SigninActivity.this, HomeActivity.class);
                                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                                        startActivity(intent);
                                    } else {//실패했을때
                                        Toast.makeText(SigninActivity.this, "로그인 오류", Toast.LENGTH_SHORT).show();
                                    }
                                }
                            });
                }
            }
        });
        signup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), SignUpActivity.class);
                startActivity(intent);
            }
        });

    }

    public static void signOut() {
        FirebaseAuth.getInstance().signOut();
    }

    public static void AccountDelete() {
        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
        user.delete();
    }

    public void firebase_login_check(){
    FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
    if (user != null) {
        Intent intent = new Intent(SigninActivity.this, HomeActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
    }
    }
}

File: YangDo_Platform/app/src/main/java/se/jbnu/yangdoplatform/SigninActivity.kt
package se.jbnu.yangdoplatform

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseAuth.AuthStateListener
import se.jbnu.yangdoplatform.SignUpActivity

class SigninActivity : AppCompatActivity() {
    private var signup: Button? = null
    private var login: Button? = null
    private var email_login: EditText? = null
    private var pwd_login: EditText? = null
    var firebaseAuth: FirebaseAuth? = null
    private val firebaseAuthListener: AuthStateListener? = null
    private val RC_SIGN_IN = 123
    private val TAG = "mainTag"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        firebase_login_check()
        setContentView(R.layout.activity_signin)
        signup = findViewById<View>(R.id.btn_signup) as Button
        login = findViewById<View>(R.id.btn_login) as Button
        email_login = findViewById<View>(R.id.user_id_input) as EditText
        pwd_login = findViewById<View>(R.id.user_password_input) as EditText
        firebaseAuth = FirebaseAuth.getInstance()
        login!!.setOnClickListener {
            val email = email_login!!.text.toString().trim { it <= ' ' }
            val pwd = pwd_login!!.text.toString().trim { it <= ' ' }
            if (email.length > 0 && pwd.length > 0) {
                firebaseAuth!!.signInWithEmailAndPassword(email, pwd)
                        .addOnCompleteListener(this@SigninActivity) { task ->
                            Log.v("SUPERTAG", "successIn")
                            if (task.isSuccessful) { //성공했을때
                                Toast.makeText(this@SigninActivity, "로그인 성공", Toast.LENGTH_SHORT).show()
                                val intent = Intent(this@SigninActivity, HomeActivity::class.java)
                                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
                                startActivity(intent)
                            } else { //실패했을때
                                Toast.makeText(this@SigninActivity, "로그인 오류", Toast.LENGTH_SHORT).show()
                            }
                        }
            }
        }
        signup!!.setOnClickListener {
            val intent = Intent(applicationContext, SignUpActivity::class.java)
            startActivity(intent)
        }
    }

    fun firebase_login_check() {
        val user = FirebaseAuth.getInstance().currentUser
        if (user != null) {
            val intent = Intent(this@SigninActivity, HomeActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(intent)
        }
    }

    companion object {
        @JvmStatic
        fun signOut() {
            FirebaseAuth.getInstance().signOut()
        }

        @JvmStatic
        fun AccountDelete() {
            val user = FirebaseAuth.getInstance().currentUser
            user!!.delete()
        }
    }
}